src/univariate/continuous/loguniform.jl

```diff
@@ -0,0 +1,68 @@
+struct LogUniform{T<:Real} <: ContinuousUnivariateDistribution
+    a::T
+    b::T
+    LogUniform{T}(a::T, b::T) where {T <: Real} = new{T}(a, b)
+end
+
+function LogUniform(a::T, b::T; check_args=true) where {T <: Real}
+    check_args && @check_args(LogUniform, 0 < a < b)
+    LogUniform{T}(a, b)
+end
+
+function LogUniform(a::Real, b::Real)
+    a1,b1 = promote(float(a), float(b))
+    LogUniform(a1,b2)
+end
+
+function convert(::Type{LogUniform{T}}, d::LogUniform) where {T}
+    LogUniform(T(d.a), T(d.b))::LogUniform{T}
+end
+
+
+#### Parameters
+params(d::LogUniform) = (d.a, d.b)
+partype(::LogUniform{T}) where {T<:Real} = T
+
+#### Statistics
+
+function mean(d::LogUniform)
+    a = d.a; b = d.b
+    (b - a) / log(b/a)
+end
+function var(d::LogUniform)
+    a = d.a; b = d.b
+    log_ba = log(b/a)
+    (b^2 - a^2) / (2*log_ba) - ((b-a)/ log_ba)^2
+end
+
+#### Evaluation
+function pdf(d::LogUniform{T}, x) where {T}
+    a = d.a; b = d.b
+    if insupport(d, x)
+        1/(x*log(b/a))
+    else
+        zero(T)
+    end
+end
```

Comment:
This implementation (and the methods below) are not type stable since they include a branch that returns a value based solely on `T` but not on the type of `x`. You can have a look at other distributions (such as the recently updated `Uniform`) how this can be avoided. The general guideline is:
- Don't use the type parameter
- Perform all computations, even if `x` is not in the support (sometimes it is necessary to use special values for values that are not in the support; e.g., for a distribution with support on the positive real line it might be necessary to perform computations for negative `x` with `_x = zero(x)` to avoid calling `log` with negative values)
- Only return special values such as `zero(result)` or `oftype(result, -Inf)` for values outside the support at the end of the computations.

E.g., for `pdf` here the preferred implementation would be
```suggestion
function pdf(d::LogUniform, x::Real)
    a, b = params(d)
    res = inv(x * log(b / a))
    return insupport(d, x) ? res : zero(res)
end
```